src/lib/Tests/RepositoryForms/Form/Processor/PreviewFormProcessorTest.php

```diff
@@ -0,0 +1,307 @@
+    /**
+     * @param $config
+     *
+     * @return MockObject|FormInterface
+     */
+    private function generateFormMock($config): MockObject
+    {
+        $form = $this->createMock(FormInterface::class);
+        $form
+            ->expects(self::once())
+            ->method('getConfig')
+            ->willReturn($config);
+
+        return $form;
+    }
+
+    /**
+     * @param APIContent $contentDraft
+     * @param string $languageCode
+     * @param string $url
+     *
+     * @return MockObject
+     */
+    private function generateUrlGeneratorMock(APIContent $contentDraft, string $languageCode, string $url): MockObject
+    {
+        $urlGenerator = $this->createMock(UrlGeneratorInterface::class);
+        $urlGenerator
+            ->expects(self::once())
+            ->method('generate')
+            ->with('ezplatform.content.preview', [
+                'contentId' => $contentDraft->id,
+                'versionNo' => $contentDraft->getVersionInfo()->versionNo,
+                'languageCode' => $languageCode,
+            ])
+            ->willReturn($url);
+
+        return $urlGenerator;
+    }
+
+    /**
+     * @param APIContent $contentDraft
+     * @param string $languageCode
+     * @param string $url
+     *
+     * @return MockObject
+     */
+    private function generateUrlGeneratorForContentEditUrlMock(APIContent $contentDraft, string $languageCode, string $url): MockObject
+    {
+        $urlGenerator = $this->createMock(UrlGeneratorInterface::class);
+        $urlGenerator
+            ->expects(self::once())
+            ->method('generate')
+            ->with('ez_content_create_no_draft', [
+                'parentLocationId' => '234',
+                'contentTypeIdentifier' => $contentDraft->id,
+                'language' => $languageCode,
+            ])
+            ->willReturn($url);
+
+        return $urlGenerator;
+    }
+
+    /**
+     * @param $contentDraftId
+     * @param $languageCode
+     *
+     * @return APIContent
+     */
+    private function generateContentDraft($contentDraftId, $languageCode): APIContent
+    {
+        $contentDraft = new Content([
+            'versionInfo' => new VersionInfo(
+                [
+                    'contentInfo' => new ContentInfo(['id' => $contentDraftId, 'mainLanguageCode' => $languageCode]),
+                ]
+            ),
+        ]);
+
+        return $contentDraft;
+    }
+
+    private function generateNotificationHandlerMock()
+    {
+        $notificationHandler = $this->createMock(NotificationHandlerInterface::class);
+        $notificationHandler
+            ->expects(self::once())
+            ->method('error')
+            ->with('error_message')
+            ->willReturn('http://content_edit_url');
+
+        return $notificationHandler;
+    }
+
+    private function generateTranslatorMock()
+    {
+        $translator = $this->createMock(TranslatorInterface::class);
+        $translator
+            ->expects(self::once())
```

Comment:
I'm against creating strict mocks that are checking method calls - this is testing implementation detail which should not be in the scope of the test. From my perspective I don't see any point in mocking Translator interface and making sure it was called only once with specific message.

Reply:
I think it is important to ensure that we have the right error message, and we do it once. But I would like to hear another opinion @Nattfarinn @ViniTou

Reply:
Alright but using "expects only once" affects implementation details which in most cases ends up as failing tests even though the changes were correct. Please test only the outcome which in case of a PreviewFormProcessor might be not that explicit but it would be saving a draft if I remember correctly.